tempodb/tempodb.go

```diff
@@ -201,8 +204,70 @@ func (rw *readerWriter) WriteBlock(ctx context.Context, c WriteableBlock) error
 	return c.Write(ctx, rw.w)
 }
 
-func (rw *readerWriter) CompleteBlock(block *wal.AppendBlock, combiner common.ObjectCombiner) (*encoding.CompleteBlock, error) {
-	return block.Complete(rw.cfg.Block, rw.wal, combiner)
+// CompleteBlock iterates the given WAL block and flushes it to the TempoDB backend.
+func (rw *readerWriter) CompleteBlock(block *wal.AppendBlock, combiner common.ObjectCombiner) (*encoding.BackendBlock, error) {
+	return rw.CompleteBlockWithBackend(context.TODO(), block, combiner, rw.r, rw.w)
+}
+
+// CompleteBlock iterates the given WAL block but flushes it to the given backend instead of the default TempoDB backend. The
+// new block will have the same ID as the input block.
+func (rw *readerWriter) CompleteBlockWithBackend(ctx context.Context, block *wal.AppendBlock, combiner common.ObjectCombiner, r backend.Reader, w backend.Writer) (*encoding.BackendBlock, error) {
+	meta := block.Meta()
+	blockID := meta.BlockID
+	tenantID := meta.TenantID
+
+	// Default and nil check is primarily to make testing easier.
+	flushSize := DefaultFlushSizeBytes
+	if rw.compactorCfg != nil && rw.compactorCfg.FlushSizeBytes > 0 {
```

Comment:
will this ever be set? i think only the ingester will use this functionality which will not have a compactorCfg

Reply:
Single-binary deployments could have it, and also all unit/integration tests.